### PR TITLE
[Storage] az storage copy: Add parameter --content-type to fix#6466

### LINKED ISSUE #6466: Allow users to define content-type in batch upload
Reference:
https://github.com/Azure/azure-cli/issues/6402
https://github.com/Azure/azure-cli/issues/6257

Ideas:
1. Allow `--content-type` accept multiple file extension to content type pairs.
2. Allow `--content-type-map` accept a configure file.
3. Allow a configuration file under `.azure` folder.
4. Allow a configuration file under the source folder.

### PR DESCRIPTION
**Pull request by Juliehzl**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---
Fix #6466 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by Juliehzl** [comment]
There is an issue with azcopy 10.3 and the content-type cannot work as expected. Let's merge it when azcopy is ready.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -507,6 +507,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('exclude_path', exclude_path_type)
         c.argument('include_path', include_path_type)
         c.argument('recursive', recursive_type)
+        c.argument('content_type', arg_group='Additional Flags', help="Specify content type of the file. "
+                   "Implies no-guess-mime-type.")
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: Maybe we can use imperative mood for alignment? Also, "Implies" isn't very intuitive to me. Possibly "If unspecified, ..."?

**Comment by yonzhan** [comment]
add to S167

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: sure. updated.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -557,6 +557,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('exclude_path', exclude_path_type)
         c.argument('include_path', include_path_type)
         c.argument('recursive', recursive_type)
+        c.argument('content_type', arg_group='Additional Flags', help="Specify content type of the file. "
+                   "If unspecified, imply no-guess-mime-type.")
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: According to https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-copy#options

```suggestion
                   "If unspecified, imply --no-guess-mime-type.")
```

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: We misunderstood it! 😰 It means if `--content-type` is given, `--no-guess-mime-type` is implied.

**Pull request merged**